### PR TITLE
8327096: (fc) java/nio/channels/FileChannel/Size.java fails on partition incapable of creating large files

### PR DESCRIPTION
**Pull request by bplb**
If an `IOException` with message "File too large" is thrown while testing with a large file, then ignore the exception if the type of the `FileStore` where the file would be located is "msdos".

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8327096](https://bugs.openjdk.org/browse/JDK-8327096): (fc) java/nio/channels/FileChannel/Size.java fails on partition incapable of creating large files (**Bug** - P4)

### Reviewers
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18073/head:pull/18073` \
`$ git checkout pull/18073`

Update a local copy of the PR: \
`$ git checkout pull/18073` \
`$ git pull https://git.openjdk.org/jdk.git pull/18073/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18073`

View PR using the GUI difftool: \
`$ git pr show -t 18073`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18073.diff">https://git.openjdk.org/jdk/pull/18073.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18073#issuecomment-1972173594)

### CODE REVIEW

File: test/jdk/java/nio/channels/FileChannel/Size.java [hunk 1]
```diff
@@ -75,8 +79,15 @@ private static void testLargeFile() throws Exception {
                                          + "Expect size " + (testSize + 10)
                                          + ", actual size " + fc.size());
             }
+        } catch (IOException ioe) {
+            if ("File too large".equals(ioe.getMessage())) {
+                Path p = largeFile.toPath();
+                FileStore store = p.getFileSystem().provider().getFileStore(p);
+                if ("msdos".equals(store.type())) // FAT32
+                    return; // ignore IOE: file too big for 32 bits
```
[the remark this answers is not in the document]
Comment (bplb) [reply test/jdk/java/nio/channels/FileChannel/Size.java:87]: `SkippedException` is EE, no?

I think the main question is whether to ignore or throw.

[the remark this answers is not in the document]
Comment (djelinski) [reply test/jdk/java/nio/channels/FileChannel/Size.java:87]: I was referring to `jtreg.SkippedException`. See: https://openjdk.org/jtreg/faq.html#what-if-a-test-does-not-apply-in-a-given-situation
If you throw that, the test will still pass.

[the remark this answers is not in the document]
Comment (bplb) [reply test/jdk/java/nio/channels/FileChannel/Size.java]: So changed in eadd73312dbb2349db94d242e3b08d2d869594d4.

**Review by djelinski — APPROVED** [review]
Looks reasonable. Thanks.

**Comment by bplb** [comment]
> Looks reasonable. Thanks.

Thanks for reviewing!

**Pull request closed without merging**